### PR TITLE
Add Kani usage and verify-std section to verification book

### PR DESCRIPTION
**Pull request by jaisnan**
Adds a section in the book assuming no knowledge from the reader about Kani or the verification effort.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: doc/book.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ edit-url-template = "https://github.com/model-checking/verify-rust-std/edit/main
 runnable = false
 
 [output.linkcheck]
-
+optional = true
```
Comment (feliperodri) [inline doc/book.toml:22]: What is this for?

File: doc/src/tools/kani.md [hunk 2]
```diff
@@ -1,7 +1,124 @@
...
 
 The Kani Rust Verifier is a bit-precise model checker for Rust.
-This page will give more details on how to use Kani to verify the standard library.
-You can find more informations about how to install and use Kani in the
-[Kani book](https://model-checking.github.io/kani/).
+Kani is designed to prove safety properties in your code as well as
+the absence of undefined behavior. It uses formal methods under the hood to ensure that
+Rust programs adhere to user specified properties.
+
+You can find more informations about how to install in [this section of the Kani book](https://model-checking.github.io/kani/install-guide.html).
```
Comment (feliperodri) [inline doc/src/tools/kani.md:8]:
```suggestion
You can find more information about how to install in [this section of the Kani book](https://model-checking.github.io/kani/install-guide.html).
```

File: doc/src/tools/kani.md [hunk 3]
```diff
@@ -1,7 +1,124 @@
...
+
+VERIFICATION:- SUCCESSFUL
+Verification Time: 0.017101772s
+
+Complete - 2 successfully verified harnesses, 0 failures, 2 total.
+```
+
+## More details
+
+You can find more informations about how to install and how you can customize your use of Kani in the
```
Comment (feliperodri) [inline doc/src/tools/kani.md:123]:
```suggestion
You can find more information about how to install and how you can customize your use of Kani in the
```

File: doc/src/tools/kani.md [hunk 4]
```diff
@@ -1,7 +1,124 @@
...
+## Using Kani to verify the rust standard library
+
+To aid the std library verification effort, Kani provides a subcommand out of the box to help you get started.
+
+### Step 1
+
+Modify your local copy of the rust std library by writing proofs for the functions/methods that you want to verify.
+
+For example, insert this short blob into your copy of the std library. This blob imports the building-block APIs such as
+`assert`, `assume`, `proof` and function-contracts such as `proof_for_contract` and `fake_function`.
```
Comment (feliperodri) [inline doc/src/tools/kani.md:62]: Add a link about `function-contracts`, maybe to our RFC. We need a documentation for it.

File: doc/src/tools/kani.md [hunk 5]
```diff
@@ -1,7 +1,124 @@
...
+
+``` rust
+// src/lib/.rs
```
Comment (feliperodri) [inline doc/src/tools/kani.md:65]:
```suggestion
```

File: doc/src/tools/kani.md [hunk 6]
```diff
@@ -1,7 +1,124 @@
 # Kani Rust Verifier
 
 The Kani Rust Verifier is a bit-precise model checker for Rust.
-This page will give more details on how to use Kani to verify the standard library.
-You can find more informations about how to install and use Kani in the
-[Kani book](https://model-checking.github.io/kani/).
+Kani is designed to prove safety properties in your code as well as
```
Comment (feliperodri) [inline doc/src/tools/kani.md:4]: Add a link above to the Kani Rust Verifier.

File: doc/src/tools/kani.md [hunk 7]
```diff
@@ -1,7 +1,124 @@
...
+the absence of undefined behavior. It uses formal methods under the hood to ensure that
```
Comment (feliperodri) [inline doc/src/tools/kani.md:5]: Add a note saying that is not all undefined behavior. Do we have a documentation with a list of UBs that we can check?

Comment (feliperodri) [reply doc/src/tools/kani.md:5]: it uses "automated reasoning" under the hood... (or model checking)

File: doc/src/tools/kani.md [hunk 8]
```diff
@@ -1,7 +1,124 @@
...
+Rust programs adhere to user specified properties.
+
+You can find more informations about how to install in [this section of the Kani book](https://model-checking.github.io/kani/install-guide.html).
+
+## Usage
+
+Consider a rust function that takes an integer and returns its absolute value and
```
Comment (feliperodri) [inline doc/src/tools/kani.md:12]:
```suggestion
Consider a Rust function that takes an integer and returns its absolute value and
```
check for additional typos.

File: doc/src/tools/kani.md [hunk 9]
```diff
@@ -1,7 +1,124 @@
...
+a kani proof that invokes the function that you want to verify
```
Comment (feliperodri) [inline doc/src/tools/kani.md:13]:
```suggestion
a Kani proof that invokes the function that you want to verify
```

File: doc/src/tools/kani.md [hunk 10]
```diff
@@ -1,7 +1,124 @@
...
+
+``` rust
+// src/main.rs
```
Comment (feliperodri) [inline doc/src/tools/kani.md:16]: I don't think these comments are necessary.
```suggestion
```

File: doc/src/tools/kani.md [hunk 11]
```diff
@@ -1,7 +1,124 @@
...
+    let b = kani::any::<i32>();
+    let result = abs_diff(a, b);
+    kani::assert(result >= 0, "Result should always be more than 0");}
+```
+
+Running the command `cargo kani` in your cargo crate will give the result
+
+```
+...Metadata about the Kani run
+
```
Comment (feliperodri) [inline doc/src/tools/kani.md:38]:
```suggestion
```

File: doc/src/tools/kani.md [hunk 12]
```diff
@@ -1,7 +1,124 @@
...
+... Other properties that might have failed or succeeded.
+
+Summary:
+Verification failed for - harness
+Complete - 0 successfully verified harnesses, 1 failures, 1 total.
+```
+
+
+## Using Kani to verify the rust standard library
```
Comment (feliperodri) [inline doc/src/tools/kani.md:53]:
```suggestion
## Using Kani to verify the Rust Standard Library
```

File: doc/src/tools/kani.md [hunk 13]
```diff
@@ -1,7 +1,124 @@
...
+
+To aid the std library verification effort, Kani provides a subcommand out of the box to help you get started.
```
Comment (feliperodri) [inline doc/src/tools/kani.md:55]:
```suggestion
To aid the Rust Standard Library verification effort, Kani provides a sub-command out of the box to help you get started.
```

File: doc/src/tools/kani.md [hunk 14]
```diff
@@ -1,7 +1,124 @@
...
+
+### Step 1
+
+Modify your local copy of the rust std library by writing proofs for the functions/methods that you want to verify.
```
Comment (feliperodri) [inline doc/src/tools/kani.md:59]:
```suggestion
Modify your local copy of the Rust Standard Library by writing proofs for the functions/methods that you want to verify.
```

File: doc/src/tools/kani.md [hunk 15]
```diff
@@ -1,7 +1,124 @@
...
+
+For example, insert this short blob into your copy of the std library. This blob imports the building-block APIs such as
```
Comment (feliperodri) [inline doc/src/tools/kani.md:61]:
```suggestion
For example, insert this short blob into your copy of the library. This blob imports the building-block APIs such as
```

File: doc/src/tools/kani.md [hunk 16]
```diff
@@ -1,7 +1,124 @@
...
+}
+```
+
+### Step 2
+
+Run the following command in your local terminal:
+
+`kani verify-std -Z unstable-options "path/to/library" --target-dir "/path/to/target" -Z function-contracts -Z stubbing`.
+
+The command `kani verify-std` is a subcommand of the `kani`. This specific subcommand is used to verify the Rust standard library with the following arguments.
```
Comment (feliperodri) [inline doc/src/tools/kani.md:97]:
```suggestion
The command `kani verify-std` is a sub-command of the `kani`. This specific sub-command is used to verify the Rust Standard Library with the following arguments.
```

File: doc/src/tools/kani.md [hunk 17]
```diff
@@ -1,7 +1,124 @@
...
+
+- `"path/to/library"`: This argument specifies the path to the modified Rust standard library that was prepared earlier in the script.
```
Comment (feliperodri) [inline doc/src/tools/kani.md:99]:
```suggestion
- `"path/to/library"`: This argument specifies the path to the modified Rust Standard Library that was prepared earlier in the script.
```

File: doc/src/tools/kani.md [hunk 18]
```diff
@@ -1,7 +1,124 @@
...
+Apart from these, you can use your regular `kani-args` such as `-Z function-contracts` and `-Z stubbing` depending on your verification needs.
+For more details on Kani's features, refer to [the features section in the Kani Book](https://model-checking.github.io/kani/reference/attributes.html)
+
+### Step 3
+
+After running the command, you can expect an output that looks like this:
+
+```
+... Details about properties that failed or suceeded verifiaction
+
```
Comment (feliperodri) [inline doc/src/tools/kani.md:111]: Move this out of the code block and explain which information this code block contains.

**Review by feliperodri — CHANGES_REQUESTED** [review]

File: doc/src/tools/kani.md (hunk 7 above)
Comment (jaisnan) [reply doc/src/tools/kani.md:5]: "Some form of UB", does that work?

**Review by feliperodri — APPROVED** [review]

**Pull request merged**